### ISSUE TITLE
Labels disappearing

### ISSUE THREAD

**Issue by TomPed** [AnalyticalGraphicsInc/cesium#3730]
This was found during AnalyticalGraphicsInc/cesium-website#982. Here is a .gif of the issue: 
![label](https://cloud.githubusercontent.com/assets/11602092/13920953/f47525ee-ef4b-11e5-899d-27310fc7a5cc.gif)
And here is the [code](https://gist.github.com/TomPed/1e5bf9f54b56bdaa6e85) for this example.

Basically what is happening is when you left click, we are picking the Milk Truck and getting the position from `pickPosition`. Then we are getting the altitude component of the position and adding it as a label. The altitude is also getting logged to the console. If you try click on the roof of the truck (1190m altitude) the label gets added fine. This is also true if you click on the hood of the truck (1188m-1189m  altitude). But the second you go to add both of the labels, the labels get messed up.

**Comment by hpinkos** [comment]
This might be Mac specific.  I can't reproduce it on my machine.

**Comment by TomPed** [comment]
In that case I'm running:

OS: OS X El Capitan, Version: 10.11.3 (15D21)

Browser: Chrome, Version 49.0.2623.87 (64-bit)

**Comment by pjcozzi** [comment]
I can reproduce this on Windows.  As I discussed with @TomPed offline, I suspect this happens when the texture atlas is updated or recreated.

To reproduce this, click the top of the milk truck, then start click lower spots until you click the bottom of the wheel:

![ok](https://cloud.githubusercontent.com/assets/782098/13925449/be4cc5f4-ef5e-11e5-8813-a6a4032ec565.gif)

**Comment by pjcozzi** [comment]
@bagnell can you add this to your list?  @lilleyse and I also ran into this with picking in 3D Tiles.  If it's not a quick fix, then we can evaluate.

**Comment by pjcozzi** [comment]
I think there is also a multifrustum bounding sphere issue here:
![ok](https://cloud.githubusercontent.com/assets/782098/13934567/45c2cf68-ef89-11e5-9720-2c39f4b5c54c.gif)

**Comment by e-andersson** [comment]
I have been running into a similar issue with labels and 3D Tiles, might be what @pjcozzi mentioned above.

When picking a position (on a 3D Tile) and then placing a label there, a partial label will appear, with some glyphs missing. Should I file a separate issue for this?

Can be reproduced in the Cesium Sandcastle:
1. Open the 3D Tiles showcase
2. Append below code snippet to the existing code
3. Click somewhere

```javascript
var labels = scene.primitives.add(new Cesium.LabelCollection());

function addLabel(position, name) {
    labels.add({
      position: position,
      text: name
    });
}

document.addEventListener('click', function(event) {
    var windowPosition = new Cesium.Cartesian2(event.clientX, event.clientY);
    var worldPosition = scene.pickPosition(windowPosition);
    addLabel(worldPosition, 'Hello glyphless world');
});
```

**Comment by pjcozzi** [comment]
@e-andersson I suspect it is the same issue so let's keep it is here unless we determine otherwise.

[3D Tiles Sandcastle demo link](http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/3d-tiles/Apps/Sandcastle/index.html?src=3D%20Tiles.html&label=Showcases)

**Comment by bagnell** [comment]
This was fixed in #3823.
